app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx

```diff
@@ -90,8 +92,16 @@ const TabLabelCanSample = ({
       });
   };
 
+  const indexOfSelectedDataset = location.pathname.lastIndexOf('/');
+  const requiredPath = location.pathname.slice(indexOfSelectedDataset + 1);
+
   return workspaceId ? (
-    <Redirect to={`/ns/${getCurrentNamespace()}/wrangler-grid/${workspaceId}`} />
+    <Redirect
+      to={{
+        pathname: `/ns/${getCurrentNamespace()}/wrangler-grid/${workspaceId}`,
+        state: { from: 'Data Sources', path: requiredPath },
```

Comment:
same as below comment, move strings "Data Sources" to constants.ts specific to this component like index.tsx, styles.ts, utils.ts and constants.ts

Reply:
created constants.ts file and moved all the constants to it from index.tsx file.